staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go

```diff
@@ -164,37 +168,61 @@ func RunTestWatchFromZero(ctx context.Context, t *testing.T, store storage.Inter
 		t.Fatalf("GuaranteedUpdate failed: %v", err)
 	}
 
+	// If testing against the cacher, we allow for slow processing by allowing
+	// the updates to propagate to the watch cache.
+	if cacher {
+		time.Sleep(100 * time.Millisecond)
+	}
 	// Make sure when we watch from 0 we receive an ADDED event
 	w, err = store.Watch(ctx, key, storage.ListOptions{ResourceVersion: "0", Predicate: storage.Everything})
 	if err != nil {
 		t.Fatalf("Watch failed: %v", err)
 	}
 	testCheckResult(t, watch.Added, w, out)
-	w.Stop()
 
-	if compaction == nil {
-		t.Skip("compaction callback not provided")
-	}
-
-	// Update again
+	// Update existing object
 	out = &example.Pod{}
 	err = store.GuaranteedUpdate(ctx, key, out, true, nil, storage.SimpleUpdate(
 		func(runtime.Object) (runtime.Object, error) {
-			return &example.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "test-ns"}}, nil
+			return &example.Pod{ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "test-ns", Annotations: map[string]string{"a": "2"}}}, nil
 		}), nil)
 	if err != nil {
 		t.Fatalf("GuaranteedUpdate failed: %v", err)
 	}
 
-	// Compact previous versions
-	compaction(ctx, t, out.ResourceVersion)
+	// If testing against the cacher, we allow for slow processing by allowing
+	// the updates to propagate to the watch cache.
+	if cacher {
+		time.Sleep(100 * time.Millisecond)
```

Comment:
Same here.

Reply:
Actually - this isn't even needed, because the event will get delivered anyway (testCheckResult is waiting if it's not yet there).